src/main/java/edu/wpi/first/gradlerio/wpi/WPIPlugin.java

```diff
@@ -0,0 +1,130 @@
+package edu.wpi.first.gradlerio.wpi;
+
+import org.gradle.api.Action;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.artifacts.repositories.MavenArtifactRepository;
+import org.gradle.api.attributes.Attribute;
+
+import edu.wpi.first.deployutils.log.ETLogger;
+import edu.wpi.first.deployutils.log.ETLoggerFactory;
+import edu.wpi.first.gradlerio.wpi.dependencies.WPIDependenciesPlugin;
+import edu.wpi.first.gradlerio.wpi.dependencies.tools.WPIToolsPlugin;
+import edu.wpi.first.nativeutils.UnzipTransform;
+
+public class WPIPlugin implements Plugin<Project> {
+    public static final Attribute<String> NATIVE_ARTIFACT_FORMAT = Attribute.of("artifactType", String.class);
+    public static final String NATIVE_ARTIFACT_ZIP_TYPE = "zip";
+    public static final String NATIVE_ARTIFACT_DIRECTORY_TYPE = "gr-directory";
+
+    private ETLogger logger;
+
+    public ETLogger getLogger() {
+        return logger;
+    }
+
+    @Override
+    public void apply(Project project) {
+        // Apply transformation for JNI and simulation artifacts
+        project.getDependencies().registerTransform(UnzipTransform.class, variantTransform -> {
+            variantTransform.getFrom().attribute(NATIVE_ARTIFACT_FORMAT, NATIVE_ARTIFACT_ZIP_TYPE);
+            variantTransform.getTo().attribute(NATIVE_ARTIFACT_FORMAT, NATIVE_ARTIFACT_DIRECTORY_TYPE);
+          });
+
+        WPIExtension wpiExtension = project.getExtensions().create("wpi", WPIExtension.class, project);
+        logger = ETLoggerFactory.INSTANCE.create(this.getClass().getSimpleName());
+
+        project.getPluginManager().apply(WPIToolsPlugin.class);
+        project.getPluginManager().apply(WPIDependenciesPlugin.class);
+
+        project.getTasks().register("wpiVersions", task -> {
+            task.setGroup("GradleRIO");
+            task.setDescription("Print all versions of the wpi block");
+            task.doLast(new Action<Task>() {
+				@Override
+				public void execute(Task arg0) {
+                                //     wpiExtension.versions().each { String key, Tuple tup ->
+            //         println "${tup.first()}: ${tup[1]} (${key})"
+            //     }
```

Comment:
This no longer prints (which is what the command is supposed to do).

Reply:
Yeah thats another thing I need to fix up. I'll do that.